generators/form/index.js

```diff
@@ -168,5 +171,95 @@ module.exports = class extends Generator {
         this.props,
       );
     }
+
+    this.updateMissingJsonSchema();
+    this.updateConstants();
+  }
+
+  updateMissingJsonSchema() {
+    const filePath = './src/platform/forms/tests/forms.unit.spec.js';
+
+    if (this.props.formNumber && !this.props.usesVetsJsonSchema) {
+      const content = this.fs.read(filePath);
+
+      try {
+        const updatedContent = content.replace(
+          /(const missingFromVetsJsonSchema = \[)([\s\S]*?)(\];)/,
+          (match, start, arrayContent, end) => {
+            const newEntry = `VA_FORM_IDS.FORM_${this.props.formNumber.replace(
+              /-/g,
+              '_',
+            )},`;
+
+            if (arrayContent.includes(newEntry)) {
+              return match;
+            }
+
+            return `${start}${arrayContent.trimEnd()}\n  ${newEntry}\n${end}`;
+          },
+        );
+
+        this.fs.write(filePath, updatedContent);
+      } catch (error) {
+        this.log(chalk.red(`Could not write to ${filePath}`));
+      }
+    }
+  }
+
+  updateConstants() {
+    const filePath = './src/platform/forms/constants.js';
+
+    if (this.props.formNumber && !this.props.usesVetsJsonSchema) {
+      const content = this.fs.read(filePath);
+
+      const newFormId = `FORM_${this.props.formNumber.replace(/-/g, '_')}`;
+
+      try {
+        const updatedContent = content.replace(
+          /(export const VA_FORM_IDS = Object\.freeze\({)([\s\S]*?)(}\))/,
```

Comment:
There are a bunch more entries in the `constants.js` file. I'm not sure if we need to update them all?

- `getAllFormLinks`
- `TRACKING_PREFIXES`
- `SIP_ENABLED_FORMS` - do we even ask if the form will use SiP?
- `MY_VA_SIP_FORMS` - WIP maintained by the Authenticated Experience team

Reply:
I've added this ticket as a followup to look into these [[yeoman] update other form id options in the constants file for things like SIP ](https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1593)